nav2_mppi_controller/include/nav2_mppi_controller/models/control_sequence.hpp

```diff
@@ -40,15 +35,15 @@ struct Control
  */
 struct ControlSequence
 {
-  xt::xtensor<float, 1> vx;
-  xt::xtensor<float, 1> vy;
-  xt::xtensor<float, 1> wz;
+  Eigen::ArrayXf vx;
```

Comment:
I believe there are some Array1f, Array2f, etc definitions that would be more optimal given we know the dimensions of the matrix. Across the board, we should use explicit sizing when we know it which allows the compiler to make better optimizations

Reply:
Agreed

Reply:
Array2f means that the Array is one-dimensional and its size is fixed, which is 2. But in our case, we don't know the size of the control seq beforehand, right?

Reply:
See https://github.com/ros-navigation/navigation2/pull/4621#discussion_r1744084152